libshaderc_util/src/compiler.cc

```diff
@@ -67,6 +67,45 @@ std::pair<int, string_piece> DecodeLineDirective(string_piece directive) {
   return std::make_pair(line, directive);
 }
 
+// Writes the message contained in the diagnostic parameter to *dest. Assumes
+// the diagnostic message is reported for for a binary location.
+void OutputSpvToolsDiagnostic(spv_diagnostic diagnostic, std::string* dest) {
+  assert(!diagnostic->isTextSource);
+
+  std::ostringstream os;
+  os << diagnostic->position.index << ": " << diagnostic->error;
+  *dest = os.str();
+}
+
+// Disassembles the given binary. Returns SPV_SUCCESS and writes the
+// disassembled text to *text_or_error if successful. Otherwise, writes the
+// error message to *text_or_error.
+spv_result_t DisassembleBinary(const std::vector<uint32_t>& binary,
+                               std::string* text_or_error) {
+  auto spvtools_context = spvContextCreate();
```

Comment:
I'm afraid that won't work because `shaderc_util::Compiler` is a part of `struct shaderc_compile_options`. The option struct supports copying. We don't want the SPIRV-Tools context object to be copied around.

Instead, I have another [commit](https://github.com/antiagainst/shaderc/commit/bc0999f51dd833a189280bfe22962ef0592c3940) to wrap this implementation using a singleton, which should solve the frequent construction/destruction.